### PR TITLE
TF-2708 Allow Chrome to download PDF after view

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2708 

### How is this PR different from #2711 ?
- Browser's memory is properly cleaned up when window's closed

### Demo

https://github.com/linagora/tmail-flutter/assets/160106668/4689d5b2-b2a5-4d04-9752-57beb45a3301

**Responsive**

https://github.com/linagora/tmail-flutter/assets/160106668/81f6e223-2ad8-43cf-8f28-f76ae5d78a92

### CODE REVIEW

File: core/lib/data/network/download/download_manager.dart [hunk 1]
```diff
@@ -104,20 +110,60 @@ class DownloadManager {
...
+            iframe.src = url;
+            document.body.appendChild(iframe);
+          };
+          window.addEventListener("beforeunload", function listener(event) {
+            URL.revokeObjectURL(url);
+            window.removeEventListener("beforeunload", listener);
+          });
+        </script>
+      </html>''';
+  }
```
Comment (dab246) [inline core/lib/data/network/download/download_manager.dart:165]: Please move it to `HtmlUtils` class

**Comment by hoangdat** [comment]
- please check with browsers which not disable security 
- check responsive 
- please try to open pdf in preview, then close web app tab, then click on download

File: core/lib/data/network/download/download_manager.dart (hunk 1 above)
Comment (tddang-linagora) [reply core/lib/data/network/download/download_manager.dart]: Done

File: core/lib/data/network/download/download_manager.dart [hunk 2]
```diff
@@ -1,19 +1,25 @@
 import 'dart:async';
 import 'dart:io';
+import 'dart:js_interop';
```
[the remark this answers is not in the document]
Comment (tddang-linagora) [reply core/lib/data/network/download/download_manager.dart]: Done

**Comment by tddang-linagora** [comment]
@hoangdat 
> please check with browsers which not disable security

Without security, there's no problem
> check responsive
> please try to open pdf in preview, then close web app tab, then click on download

Attached another demo for those.

**Review by dab246 — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**